src/System.Slices/System/Buffers/OwnedMemory.cs

```diff
@@ -13,15 +14,46 @@ internal interface IKnown
 
     public abstract class OwnedMemory<T> : IDisposable, IKnown
     {
+        const long InitializedId = long.MinValue;
+        static long _nextId = InitializedId + 1;
+
         long _id;
         int _references;
 
-        static long _nextId = InitializedId + 1;
+        protected T[] _array;
+        protected IntPtr _pointer;
+        protected int _length;
+        protected int _offset;
 
-        const long InitializedId = long.MinValue;
-        
+        protected OwnedMemory(T[] array, int arrayOffset, IntPtr pointer, int length)
```

Comment:
But then when you just pass a pointer and length, you will have to create arraysegment. It's wasteful and not very intuitive.